src/DebuggerData.cpp

```diff
@@ -341,22 +338,22 @@ void Breakpoints::saveBreakpoints(QXmlStreamWriter& xml)
 
 		// type
 		switch (bp.type) {
-		case BREAKPOINT:
+		case Breakpoint::BREAKPOINT:
 			xml.writeAttribute("type", "breakpoint");
 			break;
-		case WATCHPOINT_IOREAD:
+		case Breakpoint::WATCHPOINT_IOREAD:
 			xml.writeAttribute("type", "ioread");
 			break;
-		case WATCHPOINT_IOWRITE:
+		case Breakpoint::WATCHPOINT_IOWRITE:
 			xml.writeAttribute("type", "iowrite");
 			break;
-		case WATCHPOINT_MEMREAD:
+		case Breakpoint::WATCHPOINT_MEMREAD:
 			xml.writeAttribute("type", "memread");
 			break;
-		case WATCHPOINT_MEMWRITE:
+		case Breakpoint::WATCHPOINT_MEMWRITE:
 			xml.writeAttribute("type", "memwrite");
 			break;
-		case CONDITION:
+		case Breakpoint::CONDITION:
 			xml.writeAttribute("type", "condition");
 			break;
 		}
```

Comment:
I'm very sorry to keep giving more remarks when you touch more code, because of other stuff I requested/suggested.
It says more about the current state of the code rather than the work you're doing. Thanks again for your work! So absolutely feel free to not address these extra remarks.

The only thing this switch statement does is translate `Breakpoint::Type` into a string literal and then write that as an attribute value. Wouldn't it be more clear to write it as:
```
static const char* const names[] = {
    "breakpoint", "memread", "memwrite", "ioread", "iowrite", "condition"
};
xml.writeAttribute("type", names[bp.type]);
```
I haven't examined the full code yet. Possibly there are other locations that benefit from this translation table (and in that case it might be better to place this table elsewhere instead of locally in this function).

Reply:
Actually, I think Atom is playing tricks on me. Sometimes it doesn't seems to be updating the editor when I `git checkout` or  `rebase --continue` as expected, so it ends up rewriting files that I have just checked out. It already got me confused and I had to restart it more than once. I think I will stick to vim for now.